pkg/infra/game/state/predicate.go

```diff
@@ -1,24 +1,23 @@
-package proposal
+package state
```

Comment:
Also pushing this to state seems like an unwise idea, maybe we can push it to internal of state?